include/adc.h

```diff
@@ -27,119 +27,152 @@ extern "C" {
  */
 
 /**
- * @brief ADC driver Sequence entry
- *
- * This structure defines a sequence entry used
- * to define a sample from a specific channel.
+ * @brief ADC sampling definition structure.
  */
-struct adc_seq_entry {
-	/** Clock ticks delay before sampling the ADC. */
-	s32_t sampling_delay;
-
-	/** Buffer pointer where the sample is written.*/
-	u8_t *buffer;
+struct adc_sampling {
+	/** Clock ticks delay before starting the sampling (counted from the
+	 *  start of the previous sampling in the sequence, if any). */
+	u32_t delay;
 
-	/** Length of the sampling buffer.*/
-	u32_t buffer_length;
+	/** Pointer to a buffer where the samples are to be written.
+	 *  Each sample value is sign-extended to 16 bits. */
+	s16_t *buffer;
```

Comment:
s16_t is not enough, I used external 24-bit ADC in the past. And you need more bits when doing oversampling.

Reply:
Agreed